### PR TITLE
fixed bug where get_events_all only returns last page of data

### PR DESCRIPTION
**Pull request by lilrooness**
Fix for defect - https://rally1.rallydev.com/#/29617506082d/detail/defect/54896417559

### CODE REVIEW

**Comment by tomdos** [comment]
@lilrooness Are you sure you want to prepend Accumulator to Data? Is Data a list?

**Comment by lilrooness** [comment]
@tomdos Yes, actually also happens here https://github.com/alertlogic/erlazure/pull/6/files#diff-dd6a0bf871218adbabf27e6153bb76b5L54

It was a mistake that It was not also done here before

**Comment by tomdos** [comment]
@lilrooness well, I don't think it is correct either. I meant you should have probably used [Data | Acc]

**Comment by lilrooness** [comment]
Ah, good catch

**Comment by lilrooness** [comment]
@tomdos Fixed

**Comment by tomdos** [comment]
@lilrooness can you please verify that get_events_all() really works? based on usage of get_events() in our code it seems that result is a list. in such case I am not sure that the result from get_events_all() is what you really want.

**Comment by lilrooness** [comment]
@tomdos This has now been tested locally using the AL azure account and I can confirm that this works as expected

File: src/erlazure_insights.erl [hunk 1]
```diff
@@ -49,9 +49,9 @@ follow_token(Url, DataAcc, Config) ->
     Result = request_api(Url, Config),
     case Result of
         {ok, Data, undefined} ->
-            {ok, Data};
+            {ok, lists:flatten([Data | DataAcc])};
```
Comment (lilrooness) [inline src/erlazure_insights.erl:52]: @tomdos now flattening list

**Comment by tomdos** [comment]
👍

**Pull request merged**